sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -75,6 +92,31 @@ def test_partial_dependence_regressor():
     assert pdp.shape == (1, grid_resolution)
     assert axes[0].shape[0] == grid_resolution
 
+    # Test near perfect correlation between partial dependence and diagonal
+    # when sample weights emphasize y = x predictions
+    N = 1000
+    rng = np.random.RandomState(123456)
+    X_ = np.vstack((rng.randint(2, size=(1, N)), rng.rand(N, 1).T)).T
```

Comment:
Some of the code below would be clearer if we defined just mask = randint(...).astype(bool) here, and y_ = rand (...). We can construct X_  by stacking them later.